### ISSUE TITLE
Scrollable Tabs Feature

### ISSUE THREAD

**Issue by serak** [primefaces/primeng#1701]
I have now gotten into primeng Recently and I am in love with it but one thing that still bug's me is the when you keep adding tabs it wont scroll it just start pilling up vertically.

It would be great if they have the same feature as angular material Tab Scroll

with regards
serak

**Comment by cagataycivici** [comment]
We have that in PrimeFaces, could do the same for NG as well;

**Comment by morganRotaStabelli** [comment]
It would be perfect, do you already know when it might come?

**Comment by cagataycivici** [comment]
In 4.1.1.

**Comment by serak** [comment]
now i am the happiest MAN. its would be even great if it can be scrolled on focus and scroll. 

like angular material does..

Thanks for effort in adding this feature.

**Comment by gieboyz** [comment]
in 4.1.1
still not working using scrollable="true" like https://www.primefaces.org/showcase/ui/panel/tabView.xhtml

can you give more example for this features?

**Comment by serak** [comment]
i ask for primeng and i get a response for primeface,,

what just happpend?

**Comment by lanbin45** [comment]
@gieboyz Did you try scrollable tabview in primeNG of latest version? For V4.2, the property ‘scrollable’ did not work for me.

**Comment by luboslav** [comment]
You can allow horizontal scrolling with css.

```css
.ui-tabview .ui-tabview-nav {
    white-space: nowrap;
    overflow-x: auto;
}

.ui-tabview .ui-tabview-nav li {
    display: inline-block;
    float: none;
}
```

**Comment by vittal288** [comment]
By adding the CSS also it is not working

**Comment by vittal288** [comment]
any other answer....

**Comment by rich4756** [comment]
@vittal288 The above css works. Recheck your selectors.

**Comment by vittal288** [comment]
Could you please share the working example in plunkr or any other site

**Comment by RicardoBeltrame** [comment]
Try this.

```
.ui-tabs .ui-tabs-navscroller-btn {
    	display: inline !important;
}
```

**Comment by tiagodws** [comment]
I think this issue has been forgotten @cagataycivici

**Comment by jiangyh1024** [comment]
@cagataycivici when does this functionality will be added? Does it have some difficult to implement this functionality?

**Comment by Rishi0405** [comment]
```
.ui-tabview ul.ui-tabview-nav {
        overflow: hidden;
        display: flex;
    }
```

```

```
<button (click)="scrollPrevious($event)"><< </button>
<p-tabView **#tabView** (onChange)="selectTab($event)" 
<button (click)="scrollNext($event)">>> </button>
```

```
selectTab(e) {
  e.originalEvent.target.scrollIntoView({ behavior: "smooth", block: "center", inline: "center" });
}

scrollPrevious(e) {
   this.tabView.el.nativeElement.firstElementChild.firstElementChild.scrollLeft -= 200;
}

scrollNext(e) {
  this.tabView.el.nativeElement.firstElementChild.firstElementChild.scrollLeft += 200;
}
```

**Comment by jiangyh1024** [comment]
@Rishi0405  Thanks. that's amazing. it works like a charm

**Comment by Michel73** [comment]
Please put this feature on the roadmap. I would appreciate it very much.

**Comment by cagataycivici** [comment]
Coming soon in 9.1.0.

**Comment by Agonn** [comment]
> Coming soon in 9.1.0.

@cagataycivici Hello, 9.1 came and this feature still isn't there? Or am I missing something?

**Comment by OleksandrBohuslavskyi** [comment]
So it's still not implemented, is it?

**Comment by sourdoth** [comment]
any update?

**Comment by antsteyer** [comment]
Need it here !

**Comment by OleksandrBohuslavskyi** [comment]
> Need it here !
I gave up after two days of research and made my own directive. It's pretty simple, just use [Element.scrollTo](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo)
It seems like the PremeFaces Angular team is not skilled enough to implement this feature. Sad.

**Comment by j-be** [comment]
Hi all,

I'd really love to have this feature, but I can't seem to get @Rishi0405 's solution to work in PrimeNG 12, not even if I use `p-tabview` instead of `ui-tabview`.

So I went on and created a [small StackBlitz](https://github.com/j-be/primeng-tabview-demo-wlutbg) for my current workaround (it is all in `styles.css`), but I can't seem to get the buttons to both work and look nice at the same time.

Any suggestions?

**Comment by luboslav** [comment]
> You can allow horizontal scrolling with css.
> 
> ```css
> .ui-tabview .ui-tabview-nav {
>     white-space: nowrap;
>     overflow-x: auto;
> }
> 
> .ui-tabview .ui-tabview-nav li {
>     display: inline-block;
>     float: none;
> }
> ```

I've updated css selectors to match current version of primeng.

```
.p-tabview .p-tabview-nav {
  white-space: nowrap;
  overflow-x: auto;
  overflow-y: hidden;
  display: block;
}

.p-tabview .p-tabview-nav li {
  display: inline-block;
}
```

[Stackblitz demo](https://stackblitz.com/edit/primeng-tabview-demo-isavrl?file=src/app/app.component.scss) with custom scrollbar style.

**Comment by davisb10** [comment]
This is being resolved on issue #10119

**Comment by mertsincan** [comment]
Yes, fixed in https://github.com/primefaces/primeng/issues/10119

Best Regards,
